chrome/content/zotero/overlay.js

```diff
@@ -209,7 +210,14 @@ var ZoteroOverlay = new function()
 			
 			// Make visible
 			ZoteroPane.makeVisible();
-			
+
+			// Warn about unsafe data directory on first display
+			if(Zotero.Prefs.get('warnUnsafeDataDirOnStartup') && this.warnUnsafeDataDir) {
```

Comment:
Also, might as well flip the checks, since `this.warnOnUnsafeDataDir` is dirt cheap.